mne/channels/channels.py

```diff
@@ -755,15 +755,17 @@ def drop_channels(self, ch_names):
         -----
         .. versionadded:: 0.9.0
         """
-        if isinstance(ch_names, list):
-            if not all([isinstance(ch, str) for ch in ch_names]):
-                raise ValueError("'ch_names' must be a list of strings, got "
-                                 "{}.".format([type(ch) for ch in ch_names]))
-        elif isinstance(ch_names, str):
+        if isinstance(ch_names, str):
             ch_names = [ch_names]
+
+        try:
+            all_str = all([isinstance(ch, str) for ch in ch_names])
+        except TypeError:
+            raise ValueError("'ch_names' must be iterable.")
         else:
-            raise ValueError("'ch_names' must be a list or a string, got "
-                             "{}.".format(type(ch_names)))
+            if not all_str:
```

Comment:
might as well put this conditional and the `all_str` check after the `try/except` rather than in `else` (best to keep as little as possible in the `try` block; and the purpose of the `try/except` can be isolated to "turn this into a list", thus separable from the check "are all entries in the list strings")

Reply:
Done.